### PR TITLE
fix(pdk): rework lmdb env info metrics

### PR DESCRIPTION
**Pull request by oowl**
### Summary
we have proposed the following designs instead in an effort to make the metrics more actionable:

    lmdb_capacity : lmdb_map_size / ms_psize in number of pages

    lmdb_in_use :  ms_branch_pages + ms_leaf_pages + ms_overflow_pages in number of pages

    lmdb_allocated : me_last_pgno + 1 in number of pages

    lmdb_entries : ms_entries the number of entries (key/value pairs) in the environment

We then can direct the user to use these metrics to set reasonable warning threshold and guide them on when to increase the map size. For example, if lmdb_in_use / lmdb_capacity gets above 70%, consider raising the map size.

### Checklist

- [x] The Pull Request has tests
- [x] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

FTI-5111

Sister PR: https://github.com/Kong/lua-resty-lmdb/pull/38

### CODE REVIEW

**Comment by hbagdi** [comment]
If this doesn't go in, @oowl please make sure to revert the PR that added LMDB metrics. We must not release a feature that is not ready or will have breaking changes.

**Comment by oowl** [comment]
@hbagdi  I promise that this PR will be released in version 3.4, and I will continue to push it forward when I come back from PTO in this week.

**Comment by oowl** [comment]
hold waiting for sister PR https://github.com/Kong/lua-resty-lmdb/pull/38 merged, I will change `.requirement` to `lua-resty-lmdb` bump version.

**Review by chronolaw — APPROVED** [review]
Ready to merge when lmdb version bump.

**Comment by dndx** [comment]
There are some typo in the variable name, and it needs to be fixed before merging.

File: CHANGELOG.md
Comment (chronolaw) [inline CHANGELOG.md:259]: These entry should move to `3.4.0 changed` section, not `3.3.0`.

File: CHANGELOG.md
Comment (chronolaw) [inline CHANGELOG.md:289]: Should remove this entry (in 3.3.0 section).

File: CHANGELOG.md
Comment (chronolaw) [inline CHANGELOG.md:103]: Should these move to `Changed` section? It seems not to be fixes.

**Comment by chronolaw** [comment]
Add lmdb version bump entry please.

**Review by chronolaw — APPROVED** [review]

File: CHANGELOG.md
Comment (dndx) [inline CHANGELOG.md:104]: This feature seems to be unreleased, and we should change the existing changelog instead.

**Review by dndx — CHANGES_REQUESTED** [review]

File: kong/plugins/statsd/log.lua
Comment (dndx) [inline kong/plugins/statsd/log.lua:313]: We should fix this metrics name to be consistent with the API as well.

`dbless.lmdb.in_use_bytes`
`dbless.lmdb.map_size_bytes`

Also we can expose `dbless.lmdb.max_readers` and `dbless.lmdb.num_readers` as well. More metrics is always better.

**Comment by chronolaw** [comment]
According to the discussion in https://github.com/Kong/lua-resty-lmdb/pull/38, should we close this PR?

**Comment by oowl** [comment]
Close it for https://github.com/Kong/lua-resty-lmdb/pull/38

**Pull request closed without merging**
